habitat_code/ramachandran_code_for_review.py

```diff
@@ -0,0 +1,511 @@
+print(us.states.CO.fips)
+
+colorado_tracts_url = ("https://www2.census.gov/geo/tiger/TIGER2023/"
+           "TRACT/tl_2023_08_tract.zip")
+
+colorado_tracts_path = os.path.join(data_dir, 'colorado_census_tracts.geojson')
+
+if os.path.exists(colorado_tracts_path):
+    colorado_tracts = gpd.read_file(colorado_tracts_path).to_crs("4326")
+    print("Data is already downloaded.")
+else:
+    gpd.read_file(colorado_tracts_url).to_file(colorado_tracts_path, driver="GeoJSON")
+    colorado_tracts_gdf = gpd.read_file(colorado_tracts_path).to_crs("4326")
+
+# %%
+colorado_tracts.plot(edgecolor="white", linewidth=0.5)
+
+# %%
+# Select only the census tracts in Boulder
+print(colorado_tracts.crs)
+print(boundary_gdf.crs)
+
+boulder_tracts_gdf = gpd.sjoin(colorado_tracts, boundary_gdf, how="inner", predicate="intersects")
+boulder_tracts_gdf.plot(edgecolor="white", linewidth=0.5)
+
+# Clip to city boundary 
+boulder_city_tracts_gdf = gpd.clip(boulder_tracts_gdf, boundary_gdf)
+boulder_city_tracts_gdf.plot(edgecolor="white", linewidth=0.5)
+
+# %%
+# Download census data
+
+# B06011_001E: Estimate!!Median income in the past 12 months --!!Total:	
+# Sources: https://api.census.gov/data/2019/acs/acs5/variables.html; https://pypi.org/project/census/
+
+# Obtain Census variables from the 2019 American Community Survey at the tract level for Illinois (FIPS code: 17)
+census_path = os.path.join(data_dir, 'census_data_for_colorado.csv')
+
+if os.path.exists(census_path):
+    census_df = pd.read_csv(census_path)
+    print("Data is already downloaded.")
+
+else:
+    # Authenticate
+    api_key = getpass.getpass('U.S. Census API Key')
+    c = Census(api_key)
+    c
+
+    CO_census = c.acs5.state_county_tract(fields = ('NAME', 'B06011_001E'),
+                                      state_fips = '08',
+                                      county_fips = "*",
+                                      tract = "*",
+                                      year = 2021)
+    
+    tracts_census_df = (pd.DataFrame(CO_census, 
+                                     columns=['NAME', 
+                                              'B06011_001E', 
+                                              'state', 
+                                              'county', 
+                                              'tract'])
+                        .to_csv(census_path, index=False))
+    
+    census_df = pd.read_csv(census_path)
+
+
+census_df = census_df.rename(columns={'B06011_001E': 'median_income'})
+census_df.head()
+
+# Merge census data with tracts
+
+boulder_city_tracts_gdf
+
+# %%
+# Merge datasets
+
+boulder_city_tracts_gdf = boulder_city_tracts_gdf.loc[:, ['TRACTCE', 'geometry', 'NAMELSAD']]
+
+boulder_city_tracts_gdf['TRACTCE'] = pd.to_numeric(boulder_city_tracts_gdf['TRACTCE'], errors='coerce')
+
+tracts_w_census_gdf = boulder_city_tracts_gdf.merge(census_df, left_on='TRACTCE', right_on='tract')
+
+negative_or_not = (tracts_w_census_gdf['median_income'] >= 0)
```

Comment:
I like your variable names, they're super-descriptive! Keep an eye on that line length, though, these are a bit long for my taste, which makes it harder to read.